server/src/main/java/org/opensearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -137,6 +145,7 @@ protected void clusterManagerOperation(
         final ClusterState state,
         final ActionListener<ClusterUpdateSettingsResponse> listener
     ) {
+        validateCompatibilityModeSettingRequest(request, state);
```

Comment:
lets see of we can move these validations to the settings validator itself

Reply:
Currently moving forward with validating from inside  `execute()` of the update task in `TransportClusterUpdateSettingsAction`